tests/std/tests/VSO_0180469_fill_family/test.cpp

```diff
@@ -136,4 +136,34 @@ int main() {
 
     test_uninitialized_fill(
         [](count_copies* buff, size_t n, const count_copies& src) { uninitialized_fill_n(buff, n, src); });
+
+    // Validate int is properly converted to bool
+    {
+        bool output[] = {false, true, false};
+        fill(output, output + 3, 5);
+        for (bool elem : output) {
+            assert(elem == true);

```

Comment:
These tests will likely pass even if `int` is not properly converted to `bool`. I suggest
```
for (const bool& elem : output) {
    assert(memcmp(&elem, &true_value, sizeof(bool)) == 0);
}
```

Reply:
On my PC the assertion fails when I run:
```
#include <cassert>

using namespace std;

int main() {
    bool b;
    memset(&b, 5, 1);
    assert(b == true);
}
```
And I can't use `memcmp` because it's not constexpr

Reply:
I think it depends on debug (`/Od`) vs release (`/O2`). 
Maybe in debug `== true` is literally comparison against true, and in release it is just test and branch on flags.

If that's right, then since tests run in debug, this looks enough.

Reply:
Tested this code in release width /O2 (only replaced `assert(b == true);` with `if (b != true) abort();` because `assert` does nothing in release) and it still fails.

Reply:
It mustn't be UB because this code compiles (unless it's a compiler bug):
```
#include <bit>
#include <cassert>

using namespace std;

constexpr bool f() {
    char i = 5;
    return bit_cast<bool>(i);
}

int main() {
    constexpr bool b = f();
    static_assert(b != true);
    static_assert(b != false);
}
```

Reply:
Also I found that this code compiles:
```
#include <bit>
#include <algorithm>

using namespace std;

constexpr bool f() {
    char i = 2;

    bool arr[]  = {false, true, true};
    bool arr2[] = {false, true, bit_cast<bool>(i)};
    return !equal(begin(arr), end(arr), begin(arr2));
}

int main() {
    static_assert(f());
}
```
https://github.com/microsoft/STL/blob/e000d3feaf6acaf6214aedcfa21de483931ccbb7/stl/inc/xutility#L4895
Which means that `equal` should be able to use `memcmp` with `bool`.

Reply:
Interesting.
I think it is still UB (expected behavior is a possible UB manifestation)
Usually wrong `bool`s act sometimes as true, but apparently `constexpr` evaluation does not trigger that.

Reply:
Maybe `assert(bit_cast<char>(elem) == bit_cast<char>(bool(true)))` ?

Reply:
These tests are not C++20 only so bit_cast is not always available. And even if it's UB it means that there's no reason for equal not to use `memcmp` for `bool` - not using `memcmp` in this case would make sense only if `bit_cast<bool>((char)2) == true` was always true and we know now that at least sometimes it's not.

Reply:
Also `has_unique_object_representations_v<bool>` is true which (I think) means that if two `bool`s have different representations they must compare unequal.